flytepropeller/pkg/controller/nodes/executor.go

```diff
@@ -790,6 +795,34 @@ func (c *nodeExecutor) preExecute(ctx context.Context, dag executors.DAGStructur
 	return handler.PhaseInfoNotReady("predecessor node not yet complete"), nil
 }
 
+// checkOffloadingCompat checks if the upstream and downstream nodes are compatible with the literal offloading feature
+func (c *nodeExecutor) checkOffloadingCompat(ctx context.Context, nCtx interfaces.NodeExecutionContext, inputLiterals map[string]*core.Literal, node v1alpha1.ExecutableNode) handler.PhaseInfo {
+	isOffloadLiteral := false
+	for _, val := range inputLiterals {
+		if val != nil && val.GetOffloadedMetadata() != nil {
+			isOffloadLiteral = true
+			break
+		}
+	}
+	switch node.GetKind() {
+	case v1alpha1.NodeKindTask:
+		taskID := *node.GetTaskID()
+		taskNode, err := nCtx.ExecutionContext().GetTask(taskID)
+		if err != nil {
+			return handler.PhaseInfoFailure(core.ExecutionError_SYSTEM, "GetTaskIDFailure", err.Error(), nil)
+		}
+		runtimeData := taskNode.CoreTask().GetMetadata().GetRuntime()
+		if isOffloadLiteral && (c.literalOffloadingConfig.Enabled && !c.literalOffloadingConfig.IsSupportedSDKVersion(runtimeData.GetType().String(), runtimeData.GetVersion())) {
```

Comment:
when we have an offloaded literal but the config isn't set does that mean we silently give up attempting to check compatibility? worth at least debug logging in this scenario?

Reply:
Adding error when the config is off and the consuming a offloaded output so that its clear in the UI